version: "3.7"

services:
  shiny-proxy:
    restart: on-failure
    image: openanalytics/shinyproxy:3.1.0
    networks:
      - bb-net
    expose:
      - 8080
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock

      - ${PWD}/services/shiny-proxy/application.yml:/opt/shinyproxy/application.yml
      - ${PWD}/secrets.txt:/opt/shinyproxy/env.txt
    user: root

  nginx:
    image: nginx
    restart: always
    networks:
      - bb-net
    ports:
      - 80:80
    volumes:
      - ${PWD}/services/nginx/nginx.conf:/etc/nginx/nginx.conf
      - ./services/nginx/landing:/www/landing
      - ./services/nginx/logs:/nginx/logs
    depends_on:
      - shiny-proxy
    command: [nginx-debug, '-g', 'daemon off;']

networks:
  bb-net:
